### DISCUSSION TITLE
How to use different agent (corporate proxy) per client ip ?

### DISCUSSION THREAD

**Discussion by jaredxx1** [chimurai/http-proxy-middleware#837] [General]
Hello,

I am wondering if there is any way to set the agent (http/https/socks) in relation with the client ip address.

`function createAgent(req) {`
`const ip = req.headers['x-real-ip']`
`const proxy = getClientProxy(ip) // get client specific proxy from database`
`return new HttpsProxyAgent(proxy)`
`}`

**Comment by alanhe421** [comment]
has any way?
